auth0/token.go

```diff
@@ -80,32 +80,31 @@ func (a *ClientProvider) GetToken() (string, error) {
 	}
 
 	if authToken == "" || err != nil {
-		authToken, err = a.generateToken()
+		authToken, err = a.refreshCachedToken()
 		if err != nil {
 			return "", err
 		}
-		err := a.createAuthToken(authToken)
 
 		return authToken, err
 	}
+
 	// check token validity
-	ok, err := a.isValid(authToken)
+	ok, claims, err := a.isValid(authToken)
 	if ok {
+		go func() {
+			// refresh token before expiry by X minutes
+			if claims.VerifyExpiresAt(time.Now().Add(5*time.Minute).Unix(), false) == true {
```

Comment:
I would do just 1 minute instead of 5.

Reply:
it was Fayaz request

Reply:
yes but 5/60 is almost 10% of the token lifetime, while the problem probably happens when the library returns a still valid token and then we attempt to use it (so this is a few seconds at most). I would prefer 1 minute...